### ISSUE TITLE
Ignoring `import-private-name (PLC2701)` for first party module imports

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#16694]
### Summary

I recently reran Ruff in preview mode on the typeshed codebase. And noticed that [import-private-name (PLC2701)](https://docs.astral.sh/ruff/rules/import-private-name/#import-private-name-plc2701) whines about importing private first-party modules.

That rule can still make sense as a best-practice to avoid importing private *symbols* from first-party modules, but the point of private modules is that they should be imported in the same project/package that defines them right?

In typeshed's case, we have:
```toml
[tool.ruff.lint.isort]
known-first-party = ["_utils", "ts_utils"]
```

Yet PLC2701 triggers on lines like `from _utils import MYPY_PROTOBUF_VERSION, download_file, extract_archive, run_protoc`. Which I think would always be undesired.
I understand that [known-first-party](https://docs.astral.sh/ruff/settings/#lint_isort_known-first-party) is an isort rule, but:
1. Can [import-private-name (PLC2701)](https://docs.astral.sh/ruff/rules/import-private-name/#import-private-name-plc2701) ignore first-party private module imports
2. Can the logic of "what's a first-party module import", including manual additions to `known-first-party`, be shared across isort and Pylint rules?

**Comment by Avasam** [comment]
> Could you share an example that is raised only in preview mode? I did a quick search and couldn't find any mention of preview behavior specific to PLC2701

I think the entire rule might be in preview?
Typeshed selects the `PLC` group. So running with the `--preview` flag just adds the rule to the selection.
